features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/model/TimelineItemActionComparator.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.features.messages.impl.actionlist.model
+
+class TimelineItemActionComparator : Comparator<TimelineItemAction> {
+    // See order in https://www.figma.com/design/ux3tYoZV9WghC7hHT9Fhk0/Compound-iOS-Components?node-id=2946-2392
+    private val orderedList = listOf(
+        TimelineItemAction.EndPoll,
+        TimelineItemAction.ViewInTimeline,
+        TimelineItemAction.Reply,
+        TimelineItemAction.ReplyInThread,
+        TimelineItemAction.Forward,
+        TimelineItemAction.Pin,
+        TimelineItemAction.Unpin,
+        TimelineItemAction.CopyLink,
+        TimelineItemAction.Edit,
+        TimelineItemAction.CopyText,
+        TimelineItemAction.AddCaption,
+        TimelineItemAction.EditCaption,
+        TimelineItemAction.CopyCaption,
+        TimelineItemAction.RemoveCaption,
+        TimelineItemAction.ViewSource,
+        TimelineItemAction.ReportContent,
+        TimelineItemAction.Redact,
+    )
+
+    override fun compare(o1: TimelineItemAction, o2: TimelineItemAction): Int {
+        val index1 = orderedList.indexOf(o1)
+        val index2 = orderedList.indexOf(o2)
+        return index1.compareTo(index2)
```

Comment:
I guess items not added to the list above by mistake will appear at the top? Index -1 will be lower than anything else.

Do we have some way to ensure the sorting is exhaustive? Using a `when` and returning index values seems like a PITA, but I can't think of anything else.

Reply:
I wrote a `when` at the beginning, but yes, it's painful, and we have to give arbitrary index. What we could do is let the app crashes if `index == -1`? I guess that if a developer is adding an action, they will test it at least once :)